### DISCUSSION TITLE
secondary upn domain invalid mail address

### DISCUSSION THREAD

**Discussion by mastertIT** [neos-sdi/adfsmfa#353] [Q&A]
we receive this error in the mfa plugin when users using a secondary domain try to do the initial registration.

both domains are part of the UPN list in Active directory.
storage uses Active directory MFA Schema.

```

Error details: user@domain.com : user@domain.com : invalid mail address for user : user@domain.com
Node name: b4a0cf60-06e3-4942-88ee-05e867a8acf8
```

the user has an email address configured in ad that matches the upn domain.
both domains are part of the same AD, and both domains are resolveable to the same AD servers.

**Comment by redhook62** [comment]
Hi,

Sorry, although this is not consistent with authentication issues (identifying a user reliably). I can only confirm that having multiple users (in different domains) with one and the same email address is quite possible.

regards

**Comment by mastertIT** [comment]
the issue isn't that the plugin not supporting multiple domains. we did manage to get users enrolled with different upn's

it's the issue that it doesn't manage to resolve the other domain via autodiscovery to populate their email for the initial email MFA.
